superset/views/core.py

```diff
@@ -771,25 +770,31 @@ def explore(  # pylint: disable=too-many-locals,too-many-return-statements
                 status=400,
             )
 
-        if action in ("saveas", "overwrite"):
+        if datasource_id and action in ("saveas", "overwrite"):
             return self.save_or_overwrite_slice(
                 slc,
                 slice_add_perm,
                 slice_overwrite_perm,
                 slice_download_perm,
                 datasource_id,
                 cast(str, datasource_type),
-                datasource.name,
+                datasource_name,
             )
 
         standalone = (
             request.args.get(utils.ReservedUrlParameters.STANDALONE.value) == "true"
         )
+        dummy_datasource_data: Dict[str, Any] = {
+            "type": datasource_type,
+            "name": datasource_name,
+            "columns": [],
+            "metrics": [],
+        }
```

Comment:
Adding a dummy datasource is easier than fixing countless places on the frontend where it's expected to exist.